### ISSUE TITLE
Automatic backend creation not happening using terragrunt while executing command - terragrunt plan-all

### ISSUE THREAD

**Issue by shubhamnegi1** [gruntwork-io/terragrunt#1351]
Hi I am trying to do the terragrunt plan-all command, I have mentioned my terragrunt.hcl with remote state configuration. But it is not happening.
It is not throwing any error just stuck on generating backend.tf file for one of the module.

-terragrunt.hcl (with backend config)
-vpc
       -terragrunt.hcl
-eks (dependency on VPC)
       -terragrunt.hcl

**Comment by brikis98** [comment]
We're going to need a lot more context to help here! Please share:

1. Your Terragrunt config
1. The command you're running
1. The full log output and error

**Comment by shubhamnegi1** [comment]
this is my root level terragrunt.hcl content:

locals {
  provider_region = "eu-central-1"
  aws_region = "eu-central-1"
}

remote_state {
  backend = "s3"
  config = {
    encrypt        = true
    bucket         = "dev-backend-bucket-poc1-negi"
    key            = "${path_relative_to_include()}/terraform.tfstate"
    region         = local.aws_region
    dynamodb_table = "terraform-locks"
  }
  generate = {
    path      = "backend.tf"
    if_exists = "overwrite_terragrunt"
  }
}

generate "provider" {
    path = "provider.tf"
    if_exists = "overwrite_terragrunt"
    contents = <<EOF
    provider "aws" {
        version = "~> 3.1"
        region= "${local.provider_region}"
    }
EOF
}


------
the command i am runnin g- terraform plan-all


---
**the log -**
PS C:\Users\z0047tzz\Desktop\modules-terragrunt\dev> terragrunt plan-all 
[terragrunt] 2020/09/16 15:45:46 Module C:\Users\z0047tzz\Desktop\modules-terragrunt\dev does not have an associated terraform configuration and will be skipped.
[terragrunt] 2020/09/16 15:45:46 Stack at C:/Users/z0047tzz/Desktop/modules-terragrunt/dev:
  => Module C:/Users/z0047tzz/Desktop/modules-terragrunt/dev/eks (excluded: false, dependencies: [C:/Users/z0047tzz/Desktop/modules-terragrunt/dev/vpc])
  => Module C:/Users/z0047tzz/Desktop/modules-terragrunt/dev/vpc (excluded: false, dependencies: [])
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:45:46 Module C:/Users/z0047tzz/Desktop/modules-terragrunt/dev/vpc must wait for 0 dependencies to finish
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:45:46 Running module C:/Users/z0047tzz/Desktop/modules-terragrunt/dev/vpc now      
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\eks] 2020/09/16 15:45:46 Module C:/Users/z0047tzz/Desktop/modules-terragrunt/dev/eks must wait for 1 dependencies to finish
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:46:05 Terraform version: 0.13.0
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:46:05 Reading Terragrunt config file at C:/Users/z0047tzz/Desktop/modules-terragrunt/dev/vpc/terragrunt.hcl
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:46:05 WARNING: no double-slash (//) found in source URL /modules/terraform/aws/vpc. Relative paths in downloaded Terraform code may not work.
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:46:05 Terraform files in C:/Users/z0047tzz/Desktop/modules-terragrunt/dev/.terragrunt-cache/t__lzulj4Sr5io1h3MBZt8q9cH0/qNFYr3Pm0VRIdw2GSDWFqGZC9RE are up to date. Will not download again.
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:46:05 Copying files from C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc into 
C:/Users/z0047tzz/Desktop/modules-terragrunt/dev/.terragrunt-cache/t__lzulj4Sr5io1h3MBZt8q9cH0/qNFYr3Pm0VRIdw2GSDWFqGZC9RE
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:46:05 Setting working directory to C:/Users/z0047tzz/Desktop/modules-terragrunt/dev/.terragrunt-cache/t__lzulj4Sr5io1h3MBZt8q9cH0/qNFYr3Pm0VRIdw2GSDWFqGZC9RE
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:46:05 The file path C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\.terragrunt-cache\t__lzulj4Sr5io1h3MBZt8q9cH0\qNFYr3Pm0VRIdw2GSDWFqGZC9RE\provider.tf already exists, but was a previously generated file by terragrunt. Since if_exists for code generation is set to "overwrite_terragrunt", regenerating file.
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:46:05 Generated file C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\.terragrunt-cache\t__lzulj4Sr5io1h3MBZt8q9cH0\qNFYr3Pm0VRIdw2GSDWFqGZC9RE\provider.tf.
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:46:05 The file path C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\.terragrunt-cache\t__lzulj4Sr5io1h3MBZt8q9cH0\qNFYr3Pm0VRIdw2GSDWFqGZC9RE\backend.tf already exists, but was a previously generated file by terragrunt. Since if_exists for code generation is set to "overwrite_terragrunt", regenerating file.
[terragrunt] [C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\vpc] 2020/09/16 15:46:05 Generated file C:\Users\z0047tzz\Desktop\modules-terragrunt\dev\.terragrunt-cache\t__lzulj4Sr5io1h3MBZt8q9cH0\qNFYr3Pm0VRIdw2GSDWFqGZC9RE\backend.tf.

**Comment by shubhamnegi1** [comment]
While running terragrunt plan command in modules , it is creating the backend. but with from the root level it is throowing this error

**Comment by brikis98** [comment]
So that's the full log output? It just exits after that? What's the exit code?

**Comment by shubhamnegi1** [comment]
Yes it remains there. it does not exit. This is complete output.
please answer this - I just want to know is it possible to have state backend generated from command- terragrunt plan-all also?
it should be.
Also is my terragrunt.hcl is correct for it syntax or prerequisite wise?

**Comment by brikis98** [comment]
> Yes it remains there. it does not exit. This is complete output.

Hm, seems like some sort of Windows bug... 

> please answer this - I just want to know is it possible to have state backend generated from command- terragrunt plan-all also?
> it should be.

Yea, `plan-all` just calls `plan` in each sub-module, and `plan`, in turn, supports all the usual backend features.

> Also is my terragrunt.hcl is correct for it syntax or prerequisite wise?

It's a bit hard to read without syntax highlighting and proper indentation. Could you  try a [fenced code block](https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks) with `hcl` as the language?

**Comment by shubhamnegi1** [comment]
`````
locals {
provider_region = "eu-central-1"
aws_region = "eu-central-1"
}

remote_state {
backend = "s3"
config = {
encrypt = true
bucket = "random-bucket-name123"
key = "${path_relative_to_include()}/terraform.tfstate"
region = local.aws_region
dynamodb_table = "terraform-locks"
}
generate = {
path = "backend.tf"
if_exists = "overwrite_terragrunt"
}
}

generate "provider" {
path = "provider.tf"
if_exists = "overwrite_terragrunt"
contents = <<EOF
provider "aws" {
version = "~> 3.1"
region= "${local.provider_region}"
}
EOF
}
`````

**Comment by orbatschow** [comment]
I can confirm this behavior with Linux (5.8.14-arch1-1).

Terragrunt: 0.25.2
Terraform: 0.13.4.

**Comment by jesse-peters** [comment]
I had the same error on a mac. I hit enter when it hung though, and I got this message:

````
[terragrunt] 2020/12/03 14:08:19 Encountered the following errors:
InvalidBucketName: The specified bucket is not valid.
```

My state bucket name is invalid in this case.
